From d3b70c1aab80fbaa4967121b04a179f1d7f9222f Mon Sep 17 00:00:00 2001
From: Mathieu Duponchelle <mathieu@centricular.com>
Date: Fri, 28 Aug 2020 17:45:48 +0200
Subject: [PATCH] vpxenc: change default for deadline to good quality

Having the deadline set to best quality causes the encoder
to be absurdly slow, most real-life users will want the good
quality tradeoff instead.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/merge_requests/707>
---
 docs/gst_plugins_cache.json | 4 ++--
 ext/vpx/gstvpxenc.c         | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/docs/gst_plugins_cache.json b/docs/gst_plugins_cache.json
index ecf6b03a0..b961886a5 100644
--- a/docs/gst_plugins_cache.json
+++ b/docs/gst_plugins_cache.json
@@ -26178,12 +26178,12 @@
                         "writable": true
                     },
                     "deadline": {
-                        "blurb": "Deadline per frame (usec, 0=disabled)",
+                        "blurb": "Deadline per frame (usec, 0=best, 1=realtime)",
                         "conditionally-available": false,
                         "construct": false,
                         "construct-only": false,
                         "controllable": false,
-                        "default": "0",
+                        "default": "1000000",
                         "max": "9223372036854775807",
                         "min": "0",
                         "mutable": "null",
diff --git a/ext/vpx/gstvpxenc.c b/ext/vpx/gstvpxenc.c
index f7b999c27..7f0b5c069 100644
--- a/ext/vpx/gstvpxenc.c
+++ b/ext/vpx/gstvpxenc.c
@@ -85,7 +85,7 @@ GST_DEBUG_CATEGORY_STATIC (gst_vpxenc_debug);
 #define DEFAULT_V_SCALING_MODE VP8E_NORMAL
 #define DEFAULT_CPU_USED 0
 #define DEFAULT_ENABLE_AUTO_ALT_REF FALSE
-#define DEFAULT_DEADLINE VPX_DL_BEST_QUALITY
+#define DEFAULT_DEADLINE VPX_DL_GOOD_QUALITY
 #define DEFAULT_NOISE_SENSITIVITY 0
 #define DEFAULT_SHARPNESS 0
 
@@ -578,7 +578,7 @@ gst_vpx_enc_class_init (GstVPXEncClass * klass)
 
   g_object_class_install_property (gobject_class, PROP_DEADLINE,
       g_param_spec_int64 ("deadline", "Deadline",
-          "Deadline per frame (usec, 0=disabled)",
+          "Deadline per frame (usec, 0=best, 1=realtime)",
           0, G_MAXINT64, DEFAULT_DEADLINE,
           (GParamFlags) (G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS |
               GST_PARAM_DOC_SHOW_DEFAULT)));
-- 
GitLab

